lib/floe/workflow/reference_path.rb

```diff
@@ -19,11 +19,11 @@ def initialize(*)
         super
 
         raise Floe::InvalidWorkflowError, "Invalid Reference Path" if payload.match?(/@|,|:|\?/)
+
         @path = JsonPath.new(payload)
                         .path[1..]
                         .map { |v| v.match(/\[(?<name>.+)\]/)["name"] }
-                        .map { |v| v[0] == "'" ? v.delete("'") : v.to_i }
-                        .compact
+                        .filter_map { |v| v[0] == "'" ? v.delete("'") : v.to_i }
```

Comment:
Nice! I didn't know filter_map was available on the versions of Ruby we supported (presuming test cases exercise this line)

Reply:
just learned this one. `rubocop` does teach us good stuff. It is ruby>=2.7